Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/DonorModule.kt

```diff
@@ -0,0 +1,32 @@
+package de.rki.coronawarnapp.datadonation.analytics.modules
+
+interface DonorModule {
+
+    suspend fun startDonation(request: Request): Contribution
+
+    /**
+     * Data that the modules may need to fullfil the request
+     */
+    interface Request
+
+    /**
+     * An object that adds the data to the protobuf container, such that the Analytics class doesn't need to know the
+     * individual data types.
+     * This also acts as a callback so the donor modules know when to discard data.
+     */
+    interface Contribution {
+        // TODO replace with protobuf `PPAData` class
+        /**
+         * You will be passed a protobuf container where the module will add it's data
+         */
+        suspend fun injectData(protobufContainer: Any)
+
+        /**
+         * This will be called with the submission result.
+         * For modules that track a data delta between each submission time:
+         * If the submission failed, i.e. network issues, the donor module may collect their data for the next attempt.
+         * If the submission was sucessful, the module may discard the data as it was submitted.
+         */
+        suspend fun finishContribution(successful: Boolean)
```

Comment:
My gut feeling would be that this should be moved outside of the `Contribution` interface - time /actual implementation will tell ...